### ISSUE TITLE
Policy: unable to apply policy dns through UI

### ISSUE THREAD

**Issue by manrodrigues** [ns1labs/orb#752]
Error when create and apply DNS policy to some agent

```
{"level":"warn","ts":1641999060.3862174,"caller":"policyMgr/manager.go:142","msg":"policy failed to apply","policy_id":"57cb2738-18ca-4b6a-8ca8-851744797965","policy_name":"policy_dns","error":"Post \"http://localhost:10853/api/v1/policies\": EOF"}
{"level":"error","ts":1641999060.3862615,"caller":"pktvisor/pktvisor.go:125","msg":"pktvisor process error","error":"signal: aborted (core dumped)","stacktrace":"github.com/ns1labs/orb/agent/backend/pktvisor.(*pktvisorBackend).checkAlive\n\t/go/src/github.com/ns1labs/orb/agent/backend/pktvisor/pktvisor.go:125\ngithub.com/ns1labs/orb/agent/backend/pktvisor.(*pktvisorBackend).GetState\n\t/go/src/github.com/ns1labs/orb/agent/backend/pktvisor/pktvisor.go:55\ngithub.com/ns1labs/orb/agent.(*orbAgent).sendSingleHeartbeat\n\t/go/src/github.com/ns1labs/orb/agent/heartbeats.go:23\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleAgentPolicies\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:35\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleGroupRPCFromCore\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:65\ngithub.com/eclipse/paho%2emqtt%2egolang.(*router).matchAndDispatch.func1.3\n\t/root/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.2.0/router.go:178\ngithub.com/eclipse/paho%2emqtt%2egolang.(*router).matchAndDispatch.func1\n\t/root/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.2.0/router.go:180"}
{"level":"error","ts":1641999060.3863297,"caller":"agent/heartbeats.go:25","msg":"failed to retrieve backend state","backend":"pktvisor","error":"signal: aborted (core dumped)","stacktrace":"github.com/ns1labs/orb/agent.(*orbAgent).sendSingleHeartbeat\n\t/go/src/github.com/ns1labs/orb/agent/heartbeats.go:25\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleAgentPolicies\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:35\ngithub.com/ns1labs/orb/agent.(*orbAgent).handleGroupRPCFromCore\n\t/go/src/github.com/ns1labs/orb/agent/rpc_from.go:65\ngithub.com/eclipse/paho%2emqtt%2egolang.(*router).matchAndDispatch.func1.3\n\t/root/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.2.0/router.go:178\ngithub.com/eclipse/paho%2emqtt%2egolang.(*router).matchAndDispatch.func1\n\t/root/go/pkg/mod/github.com/eclipse/paho.mqtt.golang@v1.2.0/router.go:180"}
```

**Comment by manrodrigues** [comment]
request:

`{"name":"policy10-dns","description":"","backend":"pktvisor","tags":{},"version":1,"policy":{"kind":"collection","input":{"tap":"default_pcap","input_type":"pcap"},"handlers":{"modules":{"handler_dns_1":{"type":"dns","config":{},"filter":{"exclude_noerror":"","only_qname_suffix":"","only_rcode":""}}}}}}`

**Comment by iatdaitan** [comment]
@gpazuch  the UI should never include a filter if it is empty.

**Comment by manrodrigues** [comment]
dns policy application is enable through UI

<br>

**Logs:**



```
"level":"info","ts":1642519705.7439945,"caller":"agent/comms.go:142","msg":"completed RPC subscription to group","name":"group1","topic":"channels/931e559b-0afd-458d-a08d-c3c3ecdcd11e/messages/fromcore"}
{"level":"warn","ts":1642519749.9600074,"caller":"pktvisor/pktvisor.go:270","msg":"scrape: no policies found, skipping"}
{"level":"info","ts":1642519749.9600592,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:09.959] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}
{"level":"info","ts":1642519770.5174167,"caller":"policyMgr/manager.go:50","msg":"managing agent policy from core","action":"manage","name":"policy1","dataset":"05df9259-3fdc-4082-86b1-7f269eb4ccc8","backend":"pktvisor","id":"5e009fef-0762-4f9b-8623-a03a767c650a","version":0}
{"level":"info","ts":1642519770.5188365,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.518] [visor] [info] policy [policy1]: parsing"}
{"level":"info","ts":1642519770.51895,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.518] [visor] [info] policy [policy1]: instantiating Tap: default_pcap"}
{"level":"info","ts":1642519770.5191083,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.519] [visor] [info] policy [policy1]: instantiating Handler handler_dns_1 of type dns"}
{"level":"info","ts":1642519770.5295956,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.529] [visor] [info] policy [policy1]: starting"}
{"level":"info","ts":1642519770.5296366,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.529] [visor] [info] policy [policy1]: starting input instance: default_pcap-policy1"}
{"level":"info","ts":1642519770.603605,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.603] [visor] [info] policy [policy1]: starting handler instance: policy1-handler_dns_1"}
{"level":"info","ts":1642519770.6042292,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:29:30.604] [visor] [info] REQUEST: POST /api/v1/policies 200"}
{"level":"info","ts":1642519770.6044471,"caller":"policyMgr/manager.go:146","msg":"policy applied successfully","policy_id":"5e009fef-0762-4f9b-8623-a03a767c650a","policy_name":"policy1"}
{"level":"info","ts":1642519809.9605744,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:30:09.960] [visor] [warning] policy1-handler_dns_1 handler for policy policy1 had a PeriodException, skipping: requested metrics period has not yet accumulated, current range is [0, 0]"}
{"level":"info","ts":1642519809.9607565,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:30:09.960] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}
{"level":"warn","ts":1642519809.9607885,"caller":"pktvisor/pktvisor.go:270","msg":"scrape: no policies found, skipping"}
{"level":"info","ts":1642519869.959361,"caller":"pktvisor/pktvisor.go:230","msg":"pktvisor stdout","log":"[2022-01-18 15:31:09.959] [visor] [info] REQUEST: GET /api/v1/policies/__all/metrics/bucket/1 200"}
{"level":"info","ts":1642519869.959586,"caller":"pktvisor/pktvisor.go:297","msg":"scraped metrics for policy","policy":"policy1","policy_id":"5e009fef-0762-4f9b-8623-a03a767c650a","payload_size_b":1032}
{"level":"info","ts":1642519869.9608898,"caller":"pktvisor/pktvisor.go:315","msg":"scraped and published metrics","topic":"channels/0f0d417d-fd02-4ee9-b937-d3287210405c/messages/be/pktvisor/m/4","payload_size_b":1032,"batch_count":1}
```

**Request:**

`{"name":"policy2","description":"","backend":"pktvisor","tags":{},"version":1,"policy":{"kind":"collection","input":{"tap":"default_pcap","input_type":"pcap"},"handlers":{"modules":{"handler_dns_1":{"type":"dns"}}}}}`

**Comment by manrodrigues** [comment]
Last commit: da9a24d1f695e6777ab0fb8eb0f283b0ff2c355d
